File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedEnum.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.EnumRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.ClassNameUtils;
import com.ulyp.ui.util.StyledText;

import java.util.Arrays;

import static com.ulyp.ui.util.CssClass.*;

public class RenderedEnum extends RenderedObject {

    public RenderedEnum(EnumRepresentation representation, RenderSettings renderSettings) {
        super(representation.getType());

        String className = renderSettings.showTypes() ? representation.getType().getName() : ClassNameUtils.toSimpleName(representation.getType().getName());

        super.getChildren().addAll(
                Arrays.asList(
                        StyledText.of(className, CALL_TREE_TYPE_NAME),
                        StyledText.of(".", CALL_TREE_PLAIN_TEXT),
                        StyledText.of(representation.getName(), CALL_TREE_PLAIN_TEXT)
                )
        );
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedEnum.kt
package com.ulyp.ui.renderers

import com.ulyp.core.recorders.EnumRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.ClassNameUtils.toSimpleName
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of

class RenderedEnum(record: EnumRecord, renderSettings: RenderSettings) : RenderedObject(record.type) {
    init {

        val className = if (renderSettings.showTypes()) record.type.name else toSimpleName(record.type.name)

        super.getChildren().addAll(
            listOf(
                of(className, CssClass.CALL_TREE_TYPE_NAME),
                of(".", CssClass.CALL_TREE_NODE_SEPARATOR),
                of(record.name, CssClass.CALL_TREE_NODE_SEPARATOR)
            )
        )
    }
}